Fix exit command in console interactive mode

boaz0 opened the pull request:
When typing "exit" the code does run "break" but since "switch" is the innermost statement and not "for", the program is still running and not exiting with status code 0.

To fix it, "return 0" when "exit" is typed instead of "break".

<!--

** Thank you for your contribution! Please read this carefully! **

Please make sure you go through the checklist below. If your PR does not meet all requirements, please file it
as a draft PR. Core team members will only review your PR once it meets all the requirements below (unless your
change is something as trivial as a typo fix).

-->

<!-- If your PR resolves an issue, please add it here. -->
Resolves # 

## Target Release

1.9.0

## Checklist

<!-- Please check of ALL items in this list for all PRs: -->

- [x] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
- [x] I have not used an AI coding assistant to create this PR.
- [x] I have written all code in this PR myself OR I have marked all code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.
- [x] I (and other contributors to this PR) have not looked at the Terraform source code while implementing this PR.

### Go checklist

<!-- If your PR contains Go code, please make sure you check off all items on this list: --> 

- [xI have run golangci-lint on my change and receive no errors relevant to my code.
- [x] I have run existing tests to ensure my code doesn't break anything.
- [ ] I have added tests for all relevant use cases of my code, and those tests are passing.
- [ ] I have only exported functions, variables and structs that should be used from other packages.
- [ ] I have added meaningful comments to all exported functions, variables, and structs.

### Website/documentation checklist

<!-- If you have changed the website, please follow this checklist: -->

- [ ] I have locally started the website as [described here](https://github.com/opentofu/opentofu/blob/main/website/README.md) and checked my changes.

ollevche commented on internal/command/console_interactive.go:
```diff
@@ -68,7 +68,7 @@ func (c *ConsoleCommand) modeInteractive(session *repl.Session, ui cli.Ui) int {
 			}
 			if exit {
-				break
+				return 0
```
nit: I would like to change other break statements to returns just in case someone puts something after the loop in the future and doesn't account for this exact return.

boaz0 commented:
@ollevche - i hope i addressed all your concerns. I ran unit tests to make sure nothing is broken and did play around to make sure the behavior is not broken.

ollevche reviewed (changes requested):
I think we shouldn't make any additional changes besides replacing `switch-case` with `if-else`. Sorry for not making it clear beforehand!

boaz0 commented:
no worries @ollevche 
I apologize for the misunderstanding on my part as well :smiley_cat:

ollevche commented on internal/command/console_interactive.go:
```diff
@@ -57,18 +57,17 @@ func (c *ConsoleCommand) modeInteractive(session *repl.Session, ui cli.Ui) int {
 			}
 			if exit {
-				break
+				return 0
```
Let's leave a break statement in there for convenience, it is not a problem since we changed `switch-case` to `if-else`

ollevche reviewed (approved):
Thank you for the fix, @boaz0!

ollevche commented:
We forgot to update changelog file for this PR and some of the others. I opened a separate PR for this: https://github.com/opentofu/opentofu/pull/2116
